src/trusted/service_runtime/sel_ldr.c

```diff
@@ -1636,39 +1645,42 @@ void NaClCopyExecutionContext(struct NaClApp *nap_parent, struct NaClApp *nap_ch
   uintptr_t tramp_pnum = NaClSysToUser(nap_parent, parent_start_addr) >> NACL_PAGESHIFT;
 
   UNREFERENCED_PARAMETER(stack_pnum_parent);
+  UNREFERENCED_PARAMETER(stack_pnum_child); //unreferenced for now, may be changed
+  UNREFERENCED_PARAMETER(stack_npages); //unreferenced for now, may be changed
 
   NaClLog(1, "stack [parent: %p] [child: %p]\n", stackaddr_parent, stackaddr_child);
   NaClLog(1, "cage_id [nap_parent: %d] [nap_child: %d]\n", nap_parent->cage_id, nap_child->cage_id);
   NaClPrintAddressSpaceLayout(nap_parent);
   NaClPrintAddressSpaceLayout(nap_child);
 
-  /* add stack mapping */
-  if (NaClMprotect(stackaddr_parent, stack_size, PROT_RW) == -1) {
-      NaClLog(LOG_FATAL, "%s\n", "parent stack address NaClMprotect failed!");
-  }
-  if (NaClMprotect(stackaddr_child, stack_size, PROT_RW) == -1) {
-      NaClLog(LOG_FATAL, "%s\n", "child stack address NaClMprotect failed!");
-  }
-  NaClVmmapAddWithOverwrite(&nap_child->mem_map,
-                            stack_pnum_child,
-                            stack_npages,
-                            PROT_RW,
-                            MAP_ANON_PRIV,
-                            NULL,
-                            0,
-                            0);
-
-  /* copy stack */
-  NaClLog(1, "Copying parent stack (%zu [%#lx] bytes) from (%p) to (%p)\n",
-          stack_size,
-          stack_size,
-          stackaddr_parent,
-          stackaddr_child);
-  memcpy(stackaddr_child, stackaddr_parent, stack_size);
-  NaClPatchAddr(nap_child->mem_start, nap_parent->mem_start, stackaddr_child, stack_size);
+  //JS: Code commented out because we may want to use and modify this if we only want to copy up to the stack pointer
```

Comment:
do you want to keep this commented still?

Reply:
Yes. Currently this is being copied with the rest of the vmmap but we may want to change this in the future.